### PR TITLE
Order functionality

### PR DESCRIPTION
**Pull request by ShastkivRuslan**
Added Order and OrderItem entities with REST endpoints

- Implemented Order and OrderItem entities with required fields.
- Created endpoints for placing an order, updating order status, and retrieving user's order history.
- Added nested endpoints to retrieve specific OrderItems within an Order.
- Endpoints include:
  - POST /api/orders to place an order.
  - GET /api/orders to get user's order history.
  - PATCH /api/orders/{id} to update order status.
  - GET /api/orders/{orderId}/items for all OrderItems in an order.
  - GET /api/orders/{orderId}/items/{itemId} for a specific OrderItem.

### CODE REVIEW

File: src/main/java/ruslan/shastkiv/bookstore/service/order/OrderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    private BigDecimal calculateTotalPrice(Set<OrderItem> orderItems) {
+        return orderItems.stream()
+                .map(orderItem -> orderItem.getPrice()
+                        .multiply(BigDecimal.valueOf(orderItem.getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+    }
+
+    private void checkIsEmptyShoppingCart(ShoppingCart userShoppingCart) {
+        if (userShoppingCart.getCartItems().isEmpty()) {
+            throw new EmptyShoppingCartException(
```
Comment (Elena-Bruyako) [inline src/main/java/ruslan/shastkiv/bookstore/service/order/OrderServiceImpl.java]: it's better to use
```suggestion
            throw new OrderProcessingException(
```

File: src/main/java/ruslan/shastkiv/bookstore/service/order/OrderServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    private final UserService userService;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderItemRepository orderItemRepository;
+
+    @Override
+    @Transactional
+    public OrderDto placeOrderByUserId(Authentication authentication,
+                                       PlaceOrderRequestDto requestDto) {
+        User user = userService.getUser(authentication);
```
Comment (Elena-Bruyako) [inline src/main/java/ruslan/shastkiv/bookstore/service/order/OrderServiceImpl.java]: why do you need `getUser` method if you have `getUserId` method
here you are using userId
`shoppingCartService.findShoppingCart(user.getId());`

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/ruslan/shastkiv/bookstore/service/order/OrderServiceImpl.java (hunk 2 above)
Comment (ShastkivRuslan) [reply src/main/java/ruslan/shastkiv/bookstore/service/order/OrderServiceImpl.java]: realy)
I think that these are echoes from the very first versions of the implementation of this method

**Comment by ShastkivRuslan** [comment]
changes after mentor review:
-removed redundant getUser() method from UserService;
-changed exception name : EmptyShoppingCartException -> OrderProcessingException;

**Review by liliia-ponomarenko — APPROVED** [review]
Awesome 😎

**Pull request merged**